### PR TITLE
Attachment

### PR DESCRIPTION
**Pull request by sondreravndal**
pass attachmentGuid and rowVersion to deleteAttachment.
Changed types. 
changed upload image function to adapt to our endpoint ''formData.append('File', selectedFile, selectedFile.name);''

### CODE REVIEW

File: src/modules/Attachments/Attachment.tsx [hunk 1]
```diff
@@ -54,7 +54,10 @@ type AttachmentProps = {
     attachment: AttachmentType | AttachmentInList;
     refreshAttachments?: React.Dispatch<React.SetStateAction<boolean>>;
     readOnly: boolean;
-    deleteAttachment?: (attachmentId: number) => Promise<void>;
+    deleteAttachment?: (
+        attachmentId?: any,
```
Comment (larskay) [inline src/modules/Attachments/Attachment.tsx:58]: string | number

File: src/modules/Attachments/UploadAttachment.tsx [hunk 2]
```diff
@@ -98,6 +98,8 @@ const UploadAttachment = ({
         setPostAttachmentStatus(AsyncStatus.LOADING);
         const formData = new FormData();
         formData.append(selectedFile.name, selectedFile);
+        formData.append('File', selectedFile, selectedFile.name);
```
Comment (larskay) [inline src/modules/Attachments/UploadAttachment.tsx:101]: Skal dette gjøres 2 ganger?

File: src/modules/Checklist copy/ChecklistV2.tsx [hunk 3]
```diff
@@ -218,12 +218,12 @@ const ChecklistV2 = ({
...
                                 }
                                 deleteAttachment={(
-                                    attachmentId: number
+                                    attachmentId: number | string
                                 ): Promise<void> =>
                                     api.deleteChecklistAttachment(
                                         plantId,
                                         checklistId,
-                                        attachmentId
+                                        attachmentId as number
```
Comment (larskay) [inline src/modules/Checklist copy/ChecklistV2.tsx:226]: i stedet for as number kan du vel bare parseInt(attachmentId) liker ikke å bruke "as Type"

File: src/modules/Checklist/Checklist.tsx [hunk 4]
```diff
@@ -233,9 +233,11 @@ const Checklist = (props: ChecklistProps): JSX.Element => {
                                     api.postChecklistAttachment(file, title)
                                 }
                                 deleteAttachment={(
-                                    attachmentId: number
+                                    attachmentId: number | string
                                 ): Promise<void> =>
-                                    api.deleteChecklistAttachment(attachmentId)
+                                    api.deleteChecklistAttachment(
+                                        attachmentId as number
```
Comment (larskay) [inline src/modules/Checklist/Checklist.tsx:239]: Samme her

File: src/modules/Attachments/UploadAttachment.tsx (hunk 2 above)
Comment (sondreravndal) [reply src/modules/Attachments/UploadAttachment.tsx:101]: nop, fixed

File: src/modules/Attachments/Attachment.tsx (hunk 1 above)
Comment (sondreravndal) [reply src/modules/Attachments/Attachment.tsx]: 👍

File: src/modules/Checklist copy/ChecklistV2.tsx (hunk 3 above)
Comment (sondreravndal) [reply src/modules/Checklist copy/ChecklistV2.tsx:226]: Får typerror på parseInt

File: src/modules/Checklist/Checklist.tsx (hunk 4 above)
Comment (sondreravndal) [reply src/modules/Checklist/Checklist.tsx:239]: Får typerror på parseInt

**Review by larskay — APPROVED** [review]

**Pull request merged**